content/articles/mock-a-datasource-for-react-using-json-server/index.md

```diff
@@ -0,0 +1,736 @@
+- [Create a React app](#create-a-react-app)
+  - [New React app](#new-react-app)
+  - [New project structure](#new-project-structure)
+  - [Header component](#header-component)
+  - [Footer component](#footer-component)
+  - [Button component](#button-component)
+  - [About page component](#about-page-component)
+  - [Budget.js file](#budgetjs-file)
+  - [AddBudget component](#addbudget-component)
+  - [Budgets.js file](#budgetsjs-file)
+- [Get the items from the server](#get-the-items-from-the-server)
+- [Style the application](#style-the-application)
+- [Install JSON Server in the machine](#install-json-server-in-the-machine)
+- [Configure the JSON Server data source in the project](#configure-the-json-server-data-source-in-the-project)
+- [Run the application](#run-the-application)
+- [More on JSON server](#more-on-json-server)
+  - [Change port, file and file paths](#change-port-file-and-file-paths)
+  - [Perform queries on the data](#perform-queries-on-the-data)
+  - [Add delay and change Host](#add-delay-and-change-host)
+- [Conclusion](#conclusion)
+- [References](#references)
+
+### Key takeaways
+By the end, the following will be learned:
+- What is the `json-server` library?
+- Set up the `json-server` library in a React.js application.
+- Connect the app to the library.
+- Using the library as a data source in place of an actual DBMS.
+- Removing the dependency from the application and discarding the prototype.
+
+### Pre-requisites
+For the article, the following are required for easy article follow-up:
+- A good IDE or text editor set up on the machine.
+- A stable internet connection.
+- React development environment already setup.
+- React development skills.
+
+Brief info on what is going to be done in the article is as follows:
+- Learn briefs on the JSON server package.
+- Create a React.js application.
+- Set it up to allow API data from links.
+- Style the application.
+- Install JSON Server in the machine.
+- Set up the JSON Server as a mock database server.
+- Configure the JSON Server data source in the project.
+- Run the application.
+
+### Understanding the json-server package
+JSON server allows almost all type of back-end requests and responses like `GET`, `POST`, `PUT`, `PATCH`, and `DELETE` methods.
+
+It has routes to access data items stored in the mock database file (JSON). Few examples are `GET /posts` for fetching all the posts, `PUT /posts/1` for updating the first post, or `DELETE /posts/1` for deleting the first post.
+
+The module allows other operations to be done on the database, such as:
+
+#### Filter
+Examples - `GET/posts?title=json-server&author=riro` or `GET/comments?author.name=riro`.
+
+These end-points filters `posts` with `title` of `json-server` and `author` name of `riro` and `comments` with the `author` name of `riro` respectively.
+
+An example is a `GET` request on `http://www.albana.com/posts?title=json-server&author=riro`.
```

Comment:
This URL doesn't work

Reply:
These were supposed to show the format of the URLs, not the actual URLs. I've corrected them

Reply:
It's better we have some mock server that responds to this request.

This URL would mislead the reader.

Probably, you can create a server that processes these requests with some sample data.